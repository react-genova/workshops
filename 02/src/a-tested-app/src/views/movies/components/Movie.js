// @flow
import * as React from 'react';
import styled from 'styled-components';
import {Container, Divider} from 'semantic-ui-react';
import type {MovieType} from './types'
import Rating from './Rating'

const ContainerStyled = styled.div`
background-color: #efefef;
border: 0.035em solid #C8C8C8;
padding: 0.5em;
width:20em;
`;

const Header = styled.div`
position:relative;
`;

const Title = styled.div`
font-size: 1.6em;
font-weight: bold;
color: #264d73;
margin-bottom:0.2em
`;

const SubTitle = styled.div`
font-size: 1.2em;
font-weight: bold;
color: #264d73;
margin-bottom:0.2em
`;

const Year = styled.div`
position:absolute;
top: 0px;
right: 0px;
`;

const Footer = styled.div`
`;

type Props = MovieType & {
    onRate:(newRate:number)=>void
};
  
const Movie = ({id, title, original, rating, year,onRate}:Props) => (
    <ContainerStyled>
        <Header>
            <Title>{title}</Title>
            <SubTitle>{original}</SubTitle>
            <Year>{year}</Year>
        </Header>
        <Divider />
        <Footer>
            <Rating rating={rating} maxRating={5} onRate={onRate}/>
        </Footer>
    </ContainerStyled>
);
export default Movie;