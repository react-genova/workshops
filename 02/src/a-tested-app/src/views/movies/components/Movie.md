### Movie ###

```js
const updateRate=(id,newRate)=>{alert(id+": "+newRate);console.log(id,newRate)};
<Movie id="1" title="Il grande Lebowski" original="The Big Lebowski" rating={5} year={1998} updateRate={updateRate}/>
```