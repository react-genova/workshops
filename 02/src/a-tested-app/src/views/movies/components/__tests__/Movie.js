//@flow
import React from 'react';
import Movie from '../Movie';
import { shallow } from 'enzyme';

describe('Movie component', () => {

    it('Movie snapshot', () => {
        const updateRate = (id:string, newRate:number)=>{};
        const component = shallow(
            <Movie id="1" title="Il grande Lebowski" original="The Big Lebowski" rating={3} year={1998} updateRate={updateRate} />
        );
        expect(component).toMatchSnapshot();
    });

});