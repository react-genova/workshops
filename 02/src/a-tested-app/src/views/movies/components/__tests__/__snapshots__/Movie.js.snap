// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie component Movie snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      Il grande Lebowski
    </styled.div>
    <styled.div>
      The Big Lebowski
    </styled.div>
    <styled.div>
      1998
    </styled.div>
  </styled.div>
  <Divider />
  <styled.div>
    <Rating
      id="1"
      maxRating={5}
      onRate={[Function]}
      rating={3}
    />
  </styled.div>
</styled.div>
`;
